rust/src/storage/s3/dynamodb_lock.rs

```diff
@@ -42,6 +45,8 @@ pub struct Options {
     pub refresh_period: Duration,
     /// The amount of time to wait in addition to `lease_duration`.
     pub additional_time_to_wait_for_lock: Duration,
+    /// If set, then `try_acquire_lock` will not fail with timeout if the active lock is non expirable.
+    pub do_not_timeout_on_non_expirable_locks: bool,
```

Comment:
under what circumstance would one want to set this to false? i am thinking whether we can just get rid of this option entirely if we don't have a use-case for it at the moment.

Reply:
Makes sense, let's not over engineer here